boto/s3/connection.py

```diff
@@ -174,9 +174,12 @@ def __init__(self, aws_access_key_id=None, aws_secret_access_key=None,
                  suppress_consec_slashes=True, anon=False,
                  validate_certs=None, profile_name=None):
         no_host_provided = False
+        # Try falling back to the boto config file's value, if present.
         if host is NoHostProvided:
-            no_host_provided = True
-            host = self.DefaultHost
+            host = boto.config.get('s3', 'host')
+            if host is None:
```

Comment:
I'm explicitly testing for the NoneType here (which is the default from get() if 'host' isn't found in the boto config).

from our style guide: "Beware of writing if x: when you really mean if x is not None:-e.g., when testing whether a variable or argument that defaults to None was set to some other value. The other value might be a value that's false in a boolean context!"